### PR TITLE
Feat: Create Group in EditMonitor page

### LINKED ISSUE #3376: How do i use Monitor Group on the add monitor as i cant find a way to create a group
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

How do i use Monitor Group on the add monitor as i cant find a way to create a group if someone could tell me where i can create this that would be great 

### 📝 Error Message(s) or Log

image

### 🐻 Uptime-Kuma Version

Latest

### 💻 Operating System and Arch

Ubuntu 20.04LTS

### 🌐 Browser

Brave latest

### 🐋 Docker Version

Latest Portainer

### 🟩 NodeJS Version

_No response_

### PR DESCRIPTION
**Pull request by chakflying**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Resolves #3376

Based on feedback from #2692 and #3376, further improve the UX of creating groups:

- Add simple dialog to create group when in the EditMonitor page.

The "Group Empty" initial beat is a bit annoying, but don't see an easy way to fix it.

## Type of change

Please delete any options that are not relevant.

- User interface (UI)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)
image
image
image
image

### CODE REVIEW

**Comment by kokofixcomputers** [comment]
Yeah that should make it better. When i first upgraded to the version of uptime kuma that allows you to do some grouping i could not figure out how to create a group. This will be more understandable and more easy

**Comment by CommanderStorm** [comment]
@chakflying could you add `Resolves #3376` to the description? ^^

**Comment by louislam** [comment]
Tested, no big issue.

However, the group is pending for the first check. Is it possible to resolve by starting the group later after the monitor has started?

image

**Comment by chakflying** [comment]
Good suggestion, I implemented it. Code is a bit ugly but works, also resolves the "Group Empty" issue.

**Comment by louislam** [comment]
It seems that the last change isn't working properly? It's a blank page for me.

image

**Comment by chakflying** [comment]
Oops I made a mistake on rebase, will fix later tonight.

**Comment by chakflying** [comment]
Should be fixed now.

File: extra/gcloud-vm.ps1 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gcloud compute instances create dns-test --project=uptime-kuma-dev --zone=asia-east2-a --machine-type=e2-medium --network-interface=network-tier=PREMIUM,nic-type=GVNIC,stack-type=IPV4_ONLY,subnet=default --no-restart-on-failure --maintenance-policy=TERMINATE --provisioning-model=SPOT --instance-termination-action=STOP --service-account=787344463394-compute@developer.gserviceaccount.com --scopes=https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append --tags=http-server,https-server --create-disk=auto-delete=yes,boot=yes,device-name=dns-test,image=projects/debian-cloud/global/images/debian-11-bullseye-v20230509,mode=rw,size=20,type=projects/uptime-kuma-dev/zones/asia-east2-a/diskTypes/pd-balanced --shielded-secure-boot --shielded-vtpm --shielded-integrity-monitoring --labels=goog-ec-src=vm_add-gcloud --reservation-affinity=any --key-revocation-action-type=stop
```
Comment (louislam) [inline extra/gcloud-vm.ps1:1]: Is it committed by mistake?

Comment (louislam) [reply extra/gcloud-vm.ps1:1]: Also extra/test-mariadb.ps1 and extra/test-push.ps1

File: src/pages/EditMonitor.vue [hunk 2]
```diff
@@ -1315,10 +1399,20 @@ message HealthCheckResponse {
...
+                    // Start the new parent monitor after edit is done
+                    if (createdNewParent) {
+                        this.startParentGroupMonitor();
+                    }
                 });
             }
         },
 
+        async startParentGroupMonitor() {
+            await sleep(2000);
```
Comment (louislam) [inline src/pages/EditMonitor.vue:1412]: I put the logic into add/edit's callback, but actually there is still a race condition if the heartbeat checking is too long. I added 2 seconds to improve the situation. I think it is acceptable as there is no callback for heartbeat.

File: extra/gcloud-vm.ps1 (hunk 1 above)
Comment (chakflying) [reply extra/gcloud-vm.ps1:1]: Oops...

File: src/pages/EditMonitor.vue (hunk 2 above)
Comment (chakflying) [reply src/pages/EditMonitor.vue:1412]: I see, yeah running after `editMonitor` is done should be better. Callbacks are making me confused.

**Pull request merged**